beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -171,7 +172,34 @@ std::pair<Sophus::SE2<Scalar>, Sophus::Matrix3<Scalar>> estimate(Poses&& poses,
 
 /// Computes mean and covariance Returns a pair consisting of the estimated mean pose and its covariance.
 /**
- * Given a range of 2D poses, computes the estimated pose by averaging the translation
+ * Given a range of poses, computes the estimated pose and the standard deviation by averaging the translation.
+ *
+ * \tparam Poses A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `std::vecotr<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Pose The pose value type of the given range.
+ * \param poses Range of 1D poses.
+ * \param weights Range of weights.
+ * \return The estimated pose and its standard deviation.
+ */
+template <
+    class Poses,
```

Comment:
@Alondruck nit: consider talking about `Scalars` and `Scalar` instead of `Poses` and `Pose`. This function could be used with any scalar quantity no matter the meaning we give to it.